opentelemetry_collector/receiver/dockerstats/scraper_test.go

```diff
@@ -78,24 +105,32 @@ func (c *fakeMetricsConsumer) ConsumeMetrics(ctx context.Context, md pdata.Metri
 	return nil
 }
 
+func fakeNow() time.Time {
+	t, _ := time.Parse(time.RFC3339, "2020-01-01T00:00:00Z")
+	return t
+}
+
 func TestScraperExport(t *testing.T) {
```

Comment:
I would like to see some additional tests for the error checking code. eg testing that the receiver correctly handles it if polling docker returns an error or malformatted json.

Reply:
Since `export()` in itself doesn't return any error, that won't be very meaningful. I added one test to ensure that errors don't break the export loop itself.

Reply:
In terms of error checking, you could unit test readResourceUsageStats and readContainerInfo which do return errors. Another option would be to check that the logs contain the correct error messages and the metrics haven't been incremented when there is an error.

Reply:
Added another case for scraper test to verify that the metric is absent for a particular container when the corresponding docker API calls for that container fails.